File: internal/resources/health.go
// Health resource — reference implementation for [P|GF].
//
// Contract:
//   - RegisterHealth(s) adds the resource definition to the MCP server.
//   - The handler receives the URI and returns resource contents + error.
package resources

import (
	"context"

	"github.com/mark3labs/mcp-go/mcp"
	"github.com/mark3labs/mcp-go/server"
)

// RegisterHealth adds the health resource to s.
// Phase 3 [P|GF]: implement this function.
func RegisterHealth(s *server.MCPServer) {
	resource := mcp.NewResource(
		"mcp://health",
		"Health Status",
		mcp.WithResourceDescription("Server health status"),
		mcp.WithMIMEType("application/json"),
	)
	s.AddResource(resource, healthHandler)
}

func healthHandler(_ context.Context, _ mcp.ReadResourceRequest) ([]mcp.ResourceContents, error) {
	// Phase 3 [P|GF]: return mcp.TextResourceContents with JSON status payload
	panic("not implemented")
}

File: src/resources/health.rs
//! Health resource — reference implementation for [P|O].
//!
//! Contract (must match the registration pattern in server.rs):
//!   - `definition()` returns the Resource metadata registered at startup.
//!   - `read()` is dispatched by URI from ServerHandler::read_resource.

use rmcp::model::{ReadResourceResult, Resource, ResourceContents};

/// Resource metadata registered in McpServer::new().
pub fn definition() -> Resource {
    Resource {
        uri: "mcp://health".into(),
        name: Some("Health Status".into()),
        description: Some("Server health status".into()),
        mime_type: Some("application/json".into()),
    }
}

/// Returns current server health as a JSON resource.
pub async fn read(_uri: &str) -> Result<ReadResourceResult, rmcp::Error> {
    Ok(ReadResourceResult {
        contents: vec![ResourceContents::text(r#"{"status": "ok"}"#, "mcp://health")],
    })
}
